src/core/toga/widgets/optioncontainer.py

```diff
@@ -3,6 +3,63 @@
 from .base import Widget
 
 
+class OptionItem:
+    def __init__(self, container, label, widget, enabled):
+        self._container = container
+        self._index = len(container)
+        self._label = label
+        self._widget = widget
+        self._enabled = enabled
+
+    @property
+    def enabled(self):
+        return self._enabled
+
+    @enabled.setter
+    def enabled(self, value):
+        self._enabled = value
+        self._container.interface._impl.set_option_enabled(self._index, value)
+
+    @property
+    def label(self):
+        return self._label
+
+    @label.setter
+    def label(self, value):
+        self._label = value
+        self._container.interface._impl.set_label(self._index, value)
+
+    def refresh(self):
+        self._widget.refresh()
+
+
+class OptionList:
+
+    def __init__(self, interface):
+        self.interface = interface
+        self._options = []
+
+    def __setitem__(self, index, option):
+        option.index = index
+        self._options[index] = option
+
+    def __getitem__(self, index):
+        return self._options[index]
+
+    def __delitem__(self, key):
+        del self._options[key]
```

Comment:
This will presumably affect the index of all the children *after* the removed index. (also - the argument should probably be index, rather than key)